services/viva-ms/lambdas/submitApplication.js

```diff
@@ -67,35 +79,63 @@ export async function main(event, context) {
   );
 
   if (vivaPostError) {
-    throw new TraceException('Failed to submit Viva application', {
-      axios: { ...vivaPostError },
-    });
+    const error = {
+      messageId: record.messageId,
+      httpStatusCode: vivaPostError.status,
+      vivaErrorCode: vivaPostError.vadaResponse?.error?.details?.errorCode ?? 'N/A',
+      vivaErrorMessage: vivaPostError.vadaResponse?.error?.details?.errorMessage ?? 'N/A',
+      caseId: SK,
+    };
+    switch (error.vivaErrorCode) {
+      case '1014':
+        log.warn(
+          'Failed to submit Viva application. Will NOT be retried.',
+          context.awsRequestId,
+          null,
+          error
+        );
+        return true;
+    }
```

Comment:
We should avoid `switch` statement. Can this be replaced with an `if` statement?

Please see this and we can have an discussion about it some day :)
https://youtu.be/7EmboKQH8lM

Reply:
fixed